src/Razor/src/Microsoft.CodeAnalysis.Razor/RefTagHelperDescriptorProvider.cs

```diff
@@ -21,6 +21,20 @@ public void Execute(TagHelperDescriptorProviderContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            var compilation = context.GetCompilation();
+            if (compilation == null)
+            {
+                return;
+            }
+
+            var elementRef = compilation.GetTypeByMetadataName(ComponentsApi.ElementRef.FullTypeName);
+            if (elementRef == null)
+            {
+                // If we can't find ElementRef, then just bail. We won't be able to compile the
+                // generated code anyway.
+                return;
+            }
+
```

Comment:
https://github.com/aspnet/AspNetCore/issues/6376

Reply:
This is an issue because it could lead to ref showing up in contexts where it doesn't work/make sense.